src/backend/resources.rs

```diff
@@ -0,0 +1,270 @@
+                        .entry((
+                            Id::from("std_mem_d3"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "std_mem_d4" => {
+                    *count_map
+                        .entry((
+                            Id::from("std_mem_d4"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d1" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d1"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d2" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d2"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d3" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d3"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d4" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d4"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ => (),
+            },
+            CellType::Component { name: _name } => todo!(),
+            _ => (),
+        }
+    }
+    count_map
+}
+
+/// Writes a CSV to stdout with primitive count information
+/// generated by `gen_count_map`.
+/// TODO (priya): fix the output format and have it output to an actual file!
+fn write_csv(count_map: HashMap<(Id, Binding), u32>) {
+    let mut wtr = csv::Writer::from_writer(io::stdout());
+    let header = vec!["Primitive", "Count", "Parameters"];
+    wtr.write_record(header).ok();
+    let mut result: Vec<u8> = vec![];
+    for ((name, params), count) in count_map {
+        result.append(&mut name.id.to_string().as_str().as_bytes().to_vec());
+        result.append(&mut (", ").as_bytes().to_vec());
+        result.append(&mut count.to_string().as_str().as_bytes().to_vec());
+        result.append(&mut (", ").as_bytes().to_vec());
+        for (id, val) in params {
+            result.append(&mut id.id.to_string().into_bytes());
+            result.append(&mut (", ").as_bytes().to_vec());
+            result.append(&mut val.to_string().into_bytes());
+            result.append(&mut (", ").as_bytes().to_vec());
```

Comment:
My gut instinct says that all these conversions to byte vectors by hand is unnecessary since a lot of theses types should already implement AsRef<[u8]>. That said it would probably take a bit more digging so I am happy to go through this together if that would be useful.